src/function.ts

```diff
@@ -61,28 +125,539 @@ export class Function<P, O> implements Integration {
+              if (Array.isArray(reversed)) {
+                tokens = [...tokens, ...reversed.map((s) => s.toString())];
+              } else {
+                tokens = [...tokens, reversed.toString()];
+              }
+            }
+          } else if (typeof obj === "object") {
+            /**
+             * Remove unnecessary fields from {@link CfnResource} that bloat or fail the closure serialization.
+             */
+            const transformCfnResource = (o: unknown): any => {
+              if (Resource.isResource(o as any)) {
+                const { node, stack, env, ...rest } = o as unknown as Resource;
+                return rest;
+              } else if (CfnResource.isCfnResource(o as any)) {
+                const {
+                  stack,
+                  node,
+                  creationStack,
+                  // don't need to serialize at runtime
+                  _toCloudFormation,
+                  // @ts-ignore - private - adds the tag manager, which we don't need
+                  cfnProperties,
+                  ...rest
+                } = transformTable(o as CfnResource);
+                return transformTaggableResource(rest);
+              } else if (Token.isUnresolved(o)) {
+                const token = (<any>o).toString();
+                // add to tokens to be turned into env variables.
+                tokens = [...tokens, token];
+                return token;
+              }
+              return o;
+            };
+
+            /**
+             * Remove unnecessary fields from {@link CfnTable} that bloat or fail the closure serialization.
+             */
+            const transformTable = (o: CfnResource): CfnResource => {
+              if (
+                o.cfnResourceType ===
+                aws_dynamodb.CfnTable.CFN_RESOURCE_TYPE_NAME
+              ) {
+                const table = o as aws_dynamodb.CfnTable;
+                if (!table.streamSpecification) {
+                  const { attrStreamArn, ...rest } = table;
+
+                  return rest as unknown as CfnResource;
+                }
+              }
+
+              return o;
+            };
+
+            /**
+             * CDK Tag manager bundles in ~200kb of junk we don't need at runtime,
+             */
+            const transformTaggableResource = (o: any) => {
+              if (TagManager.isTaggable(o)) {
+                const { tags, ...rest } = o;
+                return rest;
+              }
+              return o;
+            };
+
+            /**
+             * Remove unnecessary fields from {@link CfnTable} that bloat or fail the closure serialization.
+             */
+            const transformIntegration = (o: unknown): any => {
+              if (o && typeof o === "object" && "kind" in o) {
+                const integ = o as Integration;
+                const {
+                  appSyncVtl,
+                  asl,
+                  native: { call, preWarm } = {},
```

Comment:
We have to remember to update this every time we add a new integration type. That's probably fine because it won't happen often, right?

Reply:
We don't NEED to, but new integration types may bloat the closures. I'll make this more dynamic...

Reply:
Some typesafe runtime enforcement

 ```ts
/**
 * Maintain a typesafe runtime map of integration type keys to use elsewhere.
 * 
 * For example, removing all but native integration from the {@link Function} closure.
 */
const INTEGRATION_TYPES: Record<keyof IntegrationMethods<any>, any> = {
  appSyncVtl: true,
  asl: true,
  native: true
} 

export const INTEGRATION_TYPE_KEYS = Object.keys(INTEGRATION_TYPES) as (keyof IntegrationMethods<any>)[];
```

Reply:
slightly better

```ts
/**
 * Maintain a typesafe runtime map of integration type keys to use elsewhere.
 *
 * For example, removing all but native integration from the {@link Function} closure.
 */
const INTEGRATION_TYPES: { [P in keyof IntegrationMethods<any>]: P } = {
  appSyncVtl: "appSyncVtl",
  asl: "asl",
  native: "native",
};

export const INTEGRATION_TYPE_KEYS = Object.values(INTEGRATION_TYPES);
```